gax-httpjson/src/test/java/com/google/api/gax/httpjson/ProtoMessageResponseParserTest.java

```diff
@@ -0,0 +1,98 @@
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google LLC nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.google.api.gax.httpjson;
+
+import com.google.common.truth.Truth;
+import com.google.protobuf.Field;
+import com.google.protobuf.Field.Cardinality;
+import com.google.protobuf.Option;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ProtoMessageResponseParserTest {
+  private ProtoMessageResponseParser<Field> parser;
+  private Field field;
+  private String fieldJson;
+
+  @Before
+  public void setUp() {
+    parser =
+        ProtoMessageResponseParser.<Field>newBuilder()
+            .setDefaultInstance(Field.getDefaultInstance())
+            .build();
+
+    field =
+        Field.newBuilder()
+            .setNumber(2)
+            .setName("field_name1")
+            .addOptions(Option.newBuilder().setName("opt_name1").build())
+            .addOptions(Option.newBuilder().setName("opt_name2").build())
+            .setCardinality(Cardinality.CARDINALITY_OPTIONAL)
+            .build();
+
+    fieldJson =
+        "{\n"
+            + "  \"cardinality\": \"CARDINALITY_OPTIONAL\",\n"
+            + "  \"number\": 2,\n"
+            + "  \"name\": \"field_name1\",\n"
+            + "  \"options\": [{\n"
+            + "    \"name\": \"opt_name1\"\n"
+            + "  }, {\n"
+            + "    \"name\": \"opt_name2\"\n"
+            + "  }]\n"
+            + "}";
+  }
+
+  @Test
+  public void parse() {
+    Field actualField =
+        parser.parse(new ByteArrayInputStream(fieldJson.getBytes(StandardCharsets.UTF_8)));
+    Truth.assertThat(actualField).isEqualTo(field);
+  }
+
+  @Test
+  public void parseInvalidJson() {
+    try {
+      parser.parse(new ByteArrayInputStream("invalid".getBytes(StandardCharsets.UTF_8)));
+      Truth.assertThat(true).isFalse();
```

Comment:
Is there a way to simply assert false, preferably with a message that parsing should have failed?

Reply:
I could not find a way to do it in Truth =(. You can do it in vanilla junit, but since we are using Truth in gax, I wanted to be consistent so used truth even here. As for a comment, I think it is not necessary.

Reply:
What about `assertThrows`? Looks like Truth's authors [recommend](https://stackoverflow.com/questions/38434769/how-to-test-that-an-expected-exception-is-being-thrown-using-google-truth) sticking with vanilla JUnit for exceptions.

Reply:
Ok I just switched to vanilla `Assert.fail()` version. `assertThrows` if fine, but it seems to be designed with java8+lambdas in mind. In java7 world (that we are stuck in here) using it with the explicit internal anonymous classes looks way to gnarly.